quickwit-search/src/search_stream/leaf.rs

```diff
@@ -46,39 +55,52 @@ pub async fn leaf_search_stream(
     storage: Arc<dyn Storage>,
     splits: Vec<SplitIdAndFooterOffsets>,
     index_config: Arc<dyn IndexConfig>,
-) -> UnboundedReceiverStream<Result<LeafSearchStreamResult, tonic::Status>> {
+) -> UnboundedReceiverStream<crate::Result<LeafSearchStreamResult>> {
     let (result_sender, result_receiver) = tokio::sync::mpsc::unbounded_channel();
-    for split in splits {
-        let index_config_clone = index_config.clone();
-        let result_sender_clone = result_sender.clone();
-        let request_clone = request.clone();
-        let storage_clone = storage.clone();
-        tokio::spawn(async move {
-            let leaf_split_result = leaf_search_stream_single_split(
-                split.clone(),
-                index_config_clone,
-                &request_clone,
-                storage_clone,
-            )
-            .await
-            .map_err(SearchError::convert_to_tonic_status);
-            result_sender_clone.send(leaf_split_result).map_err(|_| {
-                SearchError::InternalError(format!(
-                    "Unable to send leaf export result for split `{}`",
-                    &split.split_id
-                ))
-            })?;
-            Result::<(), SearchError>::Ok(())
-        });
-    }
+    let request_clone = request.clone();
+    tokio::spawn(async move {
+        let mut stream =
+            leaf_search_results_stream(request_clone, storage, splits, index_config).await;
+        while let Some(item) = stream.next().await {
+            if let Err(error) = result_sender.send(item) {
```

Comment:
Just my personal preference, but I like the `map_err(...)?` pattern better.

Reply:
I prefer the `if let Err` here.

The body here is just doing logging (so a side effect).
`map_err` is more about converting an error isn't it?